packages/sdk/typescript/human-protocol-sdk/test/escrow.test.ts

```
@@ -0,0 +1,1368 @@
+
+      await expect(escrowClient.abort(escrowAddress)).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.abort).toHaveBeenCalledWith();
+    });
+  });
+
+  describe('cancel', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const invalidAddress = FAKE_ADDRESS;
+
+      await expect(escrowClient.cancel(invalidAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(false);
+
+      await expect(escrowClient.cancel(escrowAddress)).rejects.toThrow(
+        ErrorEscrowAddressIsNotProvidedByFactory
+      );
+    });
+
+    test('should successfully cancel escrow', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+
+      await escrowClient.cancel(escrowAddress);
+
+      expect(escrowClient.escrowContract.cancel).toHaveBeenCalledWith();
+    });
+
+    test('should throw an error if the cancel fails', async () => {
+      const escrowAddress = ethers.constants.AddressZero;
+
+      escrowClient.escrowFactoryContract.hasEscrow.mockReturnValue(true);
+      escrowClient.escrowContract.cancel.mockRejectedValueOnce(new Error());
+
+      await expect(escrowClient.cancel(escrowAddress)).rejects.toThrow();
+
+      expect(escrowClient.escrowContract.cancel).toHaveBeenCalledWith();
+    });
+  });
+
+  describe('abort', () => {
+    test('should throw an error if escrowAddress is an invalid address', async () => {
+      const invalidAddress = FAKE_ADDRESS;
+
+      await expect(escrowClient.abort(invalidAddress)).rejects.toThrow(
+        ErrorInvalidEscrowAddressProvided
+      );
+    });
+
+    test('should throw an error if an hasEscrow return false', async () => {
```

fix writing:
'should throw an error if hasEscrow returns false'